recipes/openvino/meta.yaml

```diff
@@ -0,0 +1,521 @@
+{% set name = "openvino" %}
+{% set version = "2022.3.0" %}
+{% set commit_number = "9052" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/openvinotoolkit/openvino/archive/refs/tags/2022.3.0.tar.gz
+    sha256: b8dc1880d9ab71bd24aa5b2565724c12fb78172613e23ce312e0b98a6d8a0dd7
+    folder: openvino_sources
+    patches:
+    - patches/001-gflags.patch
+  - url: https://github.com/openvinotoolkit/openvino_contrib/archive/13025cfb0a98a5eb0ed531ea922dcb567b879902.tar.gz  # [aarch64 or arm64]
+    sha256: a6bd13848443a2e8f15ea8c24fb049935753157d233913a2174e5cf080c98c90  # [aarch64 or arm64]
+    folder: openvino_contrib  # [aarch64 or arm64]
+  - url: https://github.com/ARM-software/ComputeLibrary/archive/1b3192e8a23513031163dc14d248f47671986121.tar.gz  # [aarch64 or arm64]
+    sha256: 7244013364f4f52341cdaed0d7995e6cd93def542388b9b0b5662e0799fc93de  # [aarch64 or arm64]
+    folder: openvino_contrib/modules/arm_plugin/thirdparty/ComputeLibrary  # [aarch64 or arm64]
+  - url: https://github.com/onnx/onnx/archive/f7ee1ac60d06abe8e26c9b6bbe1e3db5286b614b.tar.gz
+    sha256: 0624860bc300fada6c0f2b5fd18043528bd0e149aad9b3c329d478ab185f17d3
+    folder: openvino_sources/thirdparty/onnx/onnx
```

Comment:
Please explain why your package cannot use exisiting conda-forge packages instead of statically linking / redistributing source code.

i.e. onnx, ittapi, onednn, xbyak, are already available on conda-forge.

Reply:
Hi @carterbox,

Thanks for your valuable comments. Let us explain why do we use these resources:
- Xbyak (conda-forge [link](https://anaconda.org/conda-forge/xbyak/files)): 
   - looks like the package is abandoned and latest version is 5.992 while latest version is 6.69 (GitHub [link](https://github.com/herumi/xbyak/tags))
   - Secondly, xbyak package is not available on all platforms in condo-forge
   - Thirdly, xbyak is headers-only library, so we don't have any duplication here
- ONNX: conda-forge package provides Python API for ONNX, while we need ONNX C++ library which we link into one of our components. OpenVINO provides an ability to read AI models in the format of original frameworks (ONNX, Paddle, PyTorch and etc files), so we need libonnx.a to link into `openvino_onnx_frontend`. Unfortunately, conda-forge's ONNX packages does not provide this library
- OneDNN: we have our own fork of oneDNN where:
  -  a lot of other kernels are provided
  - some non-required kernels are turned off in oneDNN's kernel list
  - conda-forge's oneDNN contains both CPU and GPU kernels, while we build OpenVINO's CPU plugin we need only CPU ones.
  - oneDNN in conda-forge is compiled as dynamic library and it takes 67MB, which is quite huge. When we compile tailored oneDNN statically, our plugin takes about 35-40 MB (which also assumes a lot of code in plugin itself)
- ittAPI is available for x86 platform only, while we want arm64 as well. What is advice here?

Reply:
As a communal package ecosystem, we have to balance the removal of unused artifacts from packages and package reuse. If multiple packages rebuild the same dependency statically it wastes communal space (on the CDN and local users' package caches). Just because OpenVINO doesn't use the whole API from a third-party dependency, doesn't mean that other packages on the same end user machine don't use the other parts.

> Thirdly, xbyak is headers-only library, so we don't have any duplication here

It it's header-only. That's fine. As long as you are including the license for xbyak.

> ONNX: conda-forge package provides Python API for ONNX, while we need ONNX C++ library which we link into one of our components.

Please work with @conda-forge/onnx to add a libonnx output to their feedstock and dynamically link it.

> OneDNN: we have our own fork of oneDNN where: a lot of other kernels are provided

Do you mean to say that your fork of oneDNN contains additional code that has not been merged upstream?

> some non-required kernels are turned off ... [descibing how the oneDNN fork is optimized specifically for OpenVINO]...  oneDNN in conda-forge is compiled as dynamic library and it takes 67MB, which is quite huge. When we compile tailored oneDNN statically, our plugin takes about 35-40 MB (which also assumes a lot of code in plugin itself)

I just downloaded three linux build variants for the onednn-3.1 package. None of them are more than 10 MB compressed. Unpacked, they seem to be around 40 MB. Regardless, either your fork contains incompatible changes, so it is no longer the canonical oneDNN or this package should link to the canonical oneDNN.

> ittAPI is available for x86 platform only, while we want arm64 as well. What is advice here?

It looks like ittAPI is only a static library, so there isn't a case for library reuse at runtime. Thus, like xbyak, just make sure the license is included in the license section. In the future, if a build of a dependency isn't availble, then you should open an Issue/PR on the corresponding feedstock to have that build added. For packages with dependecies that also need to be rebuilt for additional archs, we have bot-managed migrations. https://conda-forge.org/status/

Reply:
> It it's header-only. That's fine. As long as you are including the license for xbyak.
Yep, we include all 3rd licenses that we are using
https://github.com/openvinotoolkit/openvino/blob/master/licensing/third-party-programs.txt#L406
https://github.com/artanokhov/staged-recipes/blob/openvino/recipes/openvino/meta.yaml#L508

> Please work with @conda-forge/onnx to add a libonnx output to their feedstock and dynamically link it.
Seems like ONNX team already prepared necessary changes in their feedstock but not merged yet
https://github.com/conda-forge/onnx-feedstock/pull/85

Reply:
> Do you mean to say that your fork of oneDNN contains additional code that has not been merged upstream?
Absolutely. There are tens of thousands of lines differences in hundreds of files: https://github.com/oneapi-src/oneDNN/compare/master...openvinotoolkit:oneDNN:v3.1_for_ie_master_migration

Reply:
@carterbox 
Could you please share your suggestions here, taking into account our latest comments

Reply:
If your fork of oneDNN has diverged significantly from the cannonical oneDNN, then it is not really oneDNN anymore. If you do end up merging with upstream, it would preferred that you link dynamically, but otherwise, using a subproject is fine and you should consider rebranding your fork.

Please wait for / contribute to / test with https://github.com/conda-forge/onnx-feedstock/pull/85. It looks like this PR is active (has recent activity).

Reply:
@carterbox 
we disabled ONNX Frontend for current integration and will introduce it when mentioned PR (https://github.com/conda-forge/onnx-feedstock/pull/85) will be merged

also, we started discussion regarding renaming fork of oneDNN - but this is also will take a time and I think this is not blocker right now.